src/sinks/connections/TcpClient.ts

```diff
@@ -18,14 +18,9 @@ import { LOG } from '../../utils/Logger';
 import { IEndpoint } from './IEndpoint';
 import { ISocketClient } from './ISocketClient';
 
-interface SocketExtended extends net.Socket {
-  writeable: boolean;
-  readyState: string;
-}
-
```

Comment:
We are still using `readyState` later in the code and I think this property doesn't exist on `net.Socket`. It also seems like we are reading the value of this property without ever setting it to anything, so in the switch case on line 102, I think it's always going to the default option. So it seems we either need to remove `readyState` completely or fix it so we are using is correctly.